cpp_faster_fifo/tests/comparison_tests.py

```diff
@@ -0,0 +1,173 @@
+                                        consume_many=1)
+        time_mp = process_prod_and_cons(queue_type='mp', num_producers=20, num_consumers=20, msgs_per_prod=25000,
+                                        consume_many=None)
+        self.assertLess(time_ff, time_mp)
+
+
+class ComparisonTestsWithConsumeManyAtHundred(TestCase):
+
+    def test_one_prod_to_one_cons(self):
+        time_ff = process_prod_and_cons(queue_type='ff', num_producers=1, num_consumers=1, msgs_per_prod=500000,
+                                        consume_many=100)
+        time_mp = process_prod_and_cons(queue_type='mp', num_producers=1, num_consumers=1, msgs_per_prod=500000,
+                                        consume_many=None)
+        self.assertLess(time_ff, time_mp)
+
+    def test_one_prod_to_ten_cons(self):
+        time_ff = process_prod_and_cons(queue_type='ff', num_producers=1, num_consumers=10, msgs_per_prod=500000,
+                                        consume_many=100)
+        time_mp = process_prod_and_cons(queue_type='mp', num_producers=1, num_consumers=10, msgs_per_prod=500000,
+                                        consume_many=None)
+        self.assertLess(time_ff, time_mp)
+
+    def test_ten_prod_to_one_cons(self):
+        time_ff = process_prod_and_cons(queue_type='ff', num_producers=10, num_consumers=1, msgs_per_prod=50000,
+                                        consume_many=100)
+        time_mp = process_prod_and_cons(queue_type='mp', num_producers=10, num_consumers=1, msgs_per_prod=50000,
+                                        consume_many=None)
+        self.assertLess(time_ff, time_mp)
+
+    def test_three_prod_to_twenty_cons(self):
+        time_ff = process_prod_and_cons(queue_type='ff', num_producers=3, num_consumers=20, msgs_per_prod=100000,
+                                        consume_many=100)
+        time_mp = process_prod_and_cons(queue_type='mp', num_producers=3, num_consumers=20, msgs_per_prod=100000,
+                                        consume_many=None)
+        self.assertLess(time_ff, time_mp)
+
+    def test_twenty_prod_to_three_cons(self):
+        time_ff = process_prod_and_cons(queue_type='ff', num_producers=20, num_consumers=3, msgs_per_prod=25000,
+                                        consume_many=100)
+        time_mp = process_prod_and_cons(queue_type='mp', num_producers=20, num_consumers=3, msgs_per_prod=25000,
+                                        consume_many=None)
+        self.assertLess(time_ff, time_mp)
+
+    def test_twenty_prod_to_twenty_cons(self):
+        time_ff = process_prod_and_cons(queue_type='ff', num_producers=20, num_consumers=20, msgs_per_prod=25000,
+                                        consume_many=100)
+        time_mp = process_prod_and_cons(queue_type='mp', num_producers=20, num_consumers=20, msgs_per_prod=25000,
+                                        consume_many=None)
+        self.assertLess(time_ff, time_mp)
```

Comment:
I would format it differently.
Instead of having 2 test classes let's have one class, and three function calls in each method
Two for FF one for MP.
The asserts be like this:
```
self.assertLess(time_ff, time_mp)
self.assertLess(time_ff_many, time_mp)
```

time_ff_many ought not be less than time_ff, although usually it will probably be less. Here we're just proving the point that FF is better than MP

Reply:
Done